src/main/java/Terry.java

```diff
@@ -0,0 +1,139 @@
+import task.*;
+
+import java.util.Scanner;
+import ui.Ui;
+import storage.Storage;
+
+public class Terry {
+
+    private Storage storage;
+    private Ui ui;
+
+    private static final int COMMAND_AND_DETAILS_LIMIT = 2;
+
+    public static void main(String[] args) throws TerryException {
+
+        Ui ui = new Ui();
+        Storage storage = new Storage();
+        Ui.welcomeMessage();
+
+        Scanner in = new Scanner(System.in);
+        String line;
+        UserInputList list = new UserInputList();
+        storage.loadTasksFromFile(list.getTasks());
+
+        while (true) {
+            line = in.nextLine().trim();
+            String[] words = line.split(" ", COMMAND_AND_DETAILS_LIMIT);
+            String command = words[0].toLowerCase();
+
+            if (process_command(command, list, words, line, ui, storage)) return;
+        }
+
+    }
+
+    private static boolean process_command(String command, UserInputList list, String[] words, String line, Ui ui, Storage storage) throws TerryException {
+        switch (command) {
+        case "bye":
+            Ui.showExitMessage();
+            return true;
+
+        case "list":
+            ui.printTasks(list);
+            Ui.printLine();
+            break;
+
+        case "mark":
+            handleMarkCase(words, list, ui);
+            Ui.printLine();
+            break;
+        case "delete":
+            handleDeleteCase(words, list, ui);
+            Ui.printLine();
+            break;
+
+        default:
+            try {
+                addItemToList(list, line, ui, storage);
+            } catch (TerryException e) {
+                Ui.printLine();
+            }
+            break;
+        }
+        return false;
+    }
+
+    private static void handleDeleteCase(String[] words, UserInputList list, Ui ui) throws TerryException {
+        int index_to_delete = Integer.parseInt(words[1]);
+        System.out.println("I deleted thisss: ");
+        ui.printIndexWithoutNumber(list, index_to_delete);
+        list.remove(index_to_delete);
+        ui.printListSize(list);
+    }
+
+    private static void handleMarkCase(String[] words, UserInputList list, Ui ui) throws TerryException {
+        if (words.length < COMMAND_AND_DETAILS_LIMIT) { // can also use this limit since need mark followed by index as well!
+            throw new TerryException("YOU DIDN'T PUT THE NUMBER THERE");
+        } else {
+            try {
+                int index = Integer.parseInt(words[1]);
+                if (list.getOneTask(index) != null) {
+                    list.mark(index);
+                    System.out.println("aww yea the task is marked LOOK");
+                    ui.printIndex(list, index);
+                } else {
+                    throw new TerryException("WE HAVEN'T REACHED SO MANY TASKS YET");
+                }
+            } catch (NumberFormatException e) {
+                throw new TerryException("YOU DIDN'T PUT THE NUMBER THERE");
+            }
```

Comment:
This is almost 3 levels of nesting. You might want to refactor this code to another function to improve the readability.